### PR TITLE
Refactor unit tests to use libvmi for resource creation in emptydisk

### PR DESCRIPTION
**Pull request by aliamerj**
### What this PR does
Before this PR:
- Unit tests directly created and modified VM and VMI resources by manipulating their fields.
After this PR:
- Unit tests have been refactored to use the `libvmi` package for creating and modifying VM and VMI resources, promoting readability and standardization.

Partially address: https://github.com/kubevirt/kubevirt/issues/12059

### Why we need it and why it was done in this way
 - Using the libvmi package simplifies the unit tests and makes them more readable.
 - This approach ensures a standard way of creating VM and VMI resources across tests.
 
The following tradeoffs were made:
  - Direct manipulation of resources was replaced with utility functions, which may introduce minor overhead but enhances readability and maintainability.

The following alternatives were considered:
 - Continuing with direct manipulation of resources in tests, which was deemed less maintainable.
Links to places where the discussion took place: Issue discussion: [#12059](https://github.com/kubevirt/kubevirt/issues/12059)

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE

### CODE REVIEW

**Review by fossedihelm — APPROVED** [review]
Thank you!
Could you please put `NONE` in the release note?

**Comment by fossedihelm** [comment]
@aliamerj Could you please run `make && make generate`, amend the changed and push them? Thanks

**Comment by aliamerj** [comment]
I tried `make test` and all tests pass. 
What can I do to fix it?

**Comment by fossedihelm** [comment]
> I tried `make test` and all tests pass. What can I do to fix it?

The arm unit test failures are unrelated, conformance as well. Nothing to do here :) 
Just wait for a review from an approver.
Thank you

File: pkg/emptydisk/emptydisk_test.go
Comment (xpivarc) [inline pkg/emptydisk/emptydisk_test.go:37]: This is not the same as before, it should be `libvmi.WithEmptyDisk("testdisk", "", resource.MustParse("3Gi")),`

File: pkg/emptydisk/emptydisk_test.go
Comment (xpivarc) [inline pkg/emptydisk/emptydisk_test.go:49]: Same as above

File: pkg/emptydisk/emptydisk_test.go
Comment (xpivarc) [inline pkg/emptydisk/emptydisk_test.go:64]: Same as above

**Comment by aliamerj** [comment]
@xpivarc 
fixed them

**Comment by aliamerj** [comment]
@fossedihelm 
Is there any other task I can do? Maybe something little bit challenging

**Comment by fossedihelm** [comment]
I think you need to perform again: https://github.com/kubevirt/kubevirt/pull/12229#issuecomment-2199227075

Generally speaking, always run `make && make generate`, before pushing :)

**Comment by fossedihelm** [comment]
> @fossedihelm Is there any other task I can do? Maybe something little bit challenging

I would suggest you go through the issues and check if there is something that inspires you. :) 
There is some enhancement like https://github.com/kubevirt/kubevirt/issues/11993 (just an example) that could be easy, but about the challenge you know better what fits you.
I would also suggest to participate to the next community meeting and raise the question there, and/or send an email to `kubevirt-dev` to receive a wider  visibility. WDYT?

**Review by xpivarc — COMMENTED** [review]
/approve

**Pull request merged**
